### PR TITLE
PER-725 - Add tasks to manage_cnp role for creating services to expose postgres in CloudNativePG Sandbox and EDB Postgres for K8s Sandbox in AWS, Azure and GCP

### PR DESCRIPTION
**Pull request by dougortiz**
Added support for:
- Expose Postgres and Grafana as Services
- De-Expose Postgres and Grafana as Services
- Get Cluster SuperUser Secret
- Pods: Promote and Restart
- Cluster: Reload, Restart, and Status
- Maintenance Set and Un-Set
- Renamed role from: 'manage_cnp' to 'manage_cnpg'
- Updated 'README.md' files

### CODE REVIEW

File: roles/manage_cnpg/README.md [hunk 1]
```diff
@@ -79,10 +87,10 @@ Below is an example of how to include the `manage_cnp` role:
         # Scale Replicas Number
         scale_replicas_to: 3
         # Scale Manifest Filename
-        scale_manifest_filename: roles/manage_cnp/files/cluster-example.yml
+        scale_manifest_filename: roles/manage_cnpg/files/cluster-example.yml
 
   roles:
-    - role: manage_cnp
+    - role: manage_cnpg
       when: cnp_task is defined
```
Comment (hannahms) [inline roles/manage_cnpg/README.md:94]: Did this change to `cnpg_task`?

File: manage_cnpg.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        # Cluster Grafana External Target Port
+        cluster_grafana_external_ingress_target_port: 3000
+        # Pod to promote
+        pod_to_promote: 1
+        # CloudNativePG Pod Name to Restart
+        pod_name_to_restart: cnp-sandbox-2
+
+  roles:
+    - role: manage_cnpg
+      when: cnpg_task is defined
```
Comment (hannahms) [inline manage_cnpg.yml:98]: This file also seems to appear in `/playbook-examples/manage_cnpg/manage_cnpg.yml`. The directory `/playbook-examples/manage_cnp/` also still exists. Please remove the `manage_cnp` directory from the `playbook-examples` and have this `manage_cnpg.yml` only exist in the `playbook-examples/manage_cnpg/` directory.

File: roles/manage_cnpg/files/cnp-sandbox.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Example of a Basic PostgreSQL cluster
+apiVersion: postgresql.cnpg.io/v1
+kind: Cluster
+metadata:
+  name: cnp-sandbox
```
Comment (hannahms) [inline roles/manage_cnpg/files/cnp-sandbox.yml:5]: Is this correct or should this also be `cnpg-sandbox`?

File: roles/manage_cnpg/tasks/maintenance_set.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Set Cluster to Maintenance Mode in a specific namespace
+  shell: kubectl cnpg maintenance -n {{ cnpg_namespace }} set
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/maintenance_set.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/maintenance_set.yml [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+  register: setmaintenancemode 
+
+- name: Display the results of the setting the cluster to maintenance mode
+  debug: var=setmaintenancemode.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/maintenance_set.yml:7]: `ansible.builtin.debug`

File: roles/manage_cnpg/tasks/maintenance_unset.yml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Un-Set Cluster to Maintenance Mode in a specific namespace
+  shell: kubectl cnpg maintenance -n {{ cnpg_namespace }} unset
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/maintenance_unset.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/maintenance_unset.yml [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+  register: unsetmaintenancemode 
+
+- name: Display the results of the unsetting the cluster maintenance mode process
+  debug: var=unsetmaintenancemode.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/maintenance_unset.yml:7]: `ansible.builtin.debug`

File: roles/manage_cnpg/tasks/promote_pod.yml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Promote a Cluster Pod
+  shell: kubectl cnpg promote -n {{ cnpg_namespace }} {{ cluster_name }} {{ pod_to_promote }}
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/promote_pod.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/promote_pod.yml [hunk 9]
```diff
@@ -0,0 +1,7 @@
...
+  register: promotepod
+
+- name: Display the results of the pod promotion
+  debug: var=promotepod.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/promote_pod.yml:7]: `ansible.builtin.debug`

File: roles/manage_cnpg/tasks/reload_cluster.yml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Reload a Cluster
+  shell: kubectl cnpg reload -n {{ cnpg_namespace }} {{ cluster_name }}
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/reload_cluster.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/reload_cluster.yml [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+  register: reloadcluster
+
+- name: Display the results of the cluster reload
+  debug: var=reloadcluster.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/reload_cluster.yml:7]: `ansible.builtin.debug`

File: roles/manage_cnpg/tasks/restart_cluster.yml [hunk 12]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Restart a Cluster
+  shell: kubectl cnpg restart -n {{ cnpg_namespace }} {{ cluster_name }}
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/restart_cluster.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/restart_cluster.yml [hunk 13]
```diff
@@ -0,0 +1,7 @@
...
+  register: restartcluster
+
+- name: Display the results of the cluster restart
+  debug: var=restartcluster.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/restart_cluster.yml:7]: `ansible.builtin.debug`

File: roles/manage_cnpg/tasks/restart_pod.yml [hunk 14]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Restart a Cluster Pod
+  shell: kubectl cnpg restart -n {{ cnpg_namespace }} {{ cluster_name }} {{ pod_name_to_restart }}
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/restart_pod.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/restart_pod.yml [hunk 15]
```diff
@@ -0,0 +1,7 @@
...
+  register: restartpod
+
+- name: Display the results of the pod restart
+  debug: var=restartpod.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/restart_pod.yml:7]: `ansible.builtin.debug`

File: roles/manage_cnpg/tasks/status_cluster.yml [hunk 16]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Status of a Cluster in a specific namespace
+  shell: kubectl cnpg -n {{ cnpg_namespace }} status {{ cluster_name }}
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/status_cluster.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/status_cluster.yml [hunk 17]
```diff
@@ -0,0 +1,7 @@
...
+  register: clusterstatus
+
+- name: Display the status of the cluster
+  debug: var=clusterstatus.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/status_cluster.yml:7]: `ansible.builtin.debug`

File: roles/manage_cnpg/tasks/apply_scale_manifest.yml [hunk 18]
```diff
@@ -0,0 +1,3 @@
+---
+- name: Apply the scale manifest
+  shell: kubectl scale -n {{ cnpg_namespace }} --replicas="{{ scale_replicas_to }}" -f "{{ scale_manifest_filename }}"
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/apply_scale_manifest.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/drop_expose_grafana_externally.yml [hunk 19]
```diff
@@ -0,0 +1,3 @@
+---
+- name: Drop Grafana External Service
+  shell: kubectl delete service -n {{ cnpg_namespace }} {{ cluster_grafana_external_ingress_service_name }}
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/drop_expose_grafana_externally.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/drop_expose_postgres_externally.yml [hunk 20]
```diff
@@ -0,0 +1,3 @@
+---
+- name: Drop Cluster Postgres External Service
+  shell: kubectl delete service -n {{ cnpg_namespace }} {{ cluster_external_ingress_service_name }}
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/drop_expose_postgres_externally.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/expose_grafana_externally.yml [hunk 21]
```diff
@@ -0,0 +1,3 @@
+---
+- name: Expose Grafana Deployment as an External Service
+  shell: kubectl expose -n {{ cnpg_namespace }} deploy/{{ cluster_grafana_deployment_name }} --target-port={{ cluster_grafana_external_ingress_target_port }} --port={{ cluster_grafana_external_ingress_source_port }} --name {{ cluster_grafana_external_ingress_service_name }} --type LoadBalancer
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/expose_grafana_externally.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/expose_postgres_externally.yml [hunk 22]
```diff
@@ -0,0 +1,3 @@
+---
+- name: Expose Cluster Postgres Port as an External Service
+  shell: kubectl expose -n {{ cnpg_namespace }} svc/{{ cluster_rw_service_name }} --port={{ cluster_external_ingress_source_port }} --target-port={{ cluster_external_ingress_target_port }} --name={{ cluster_external_ingress_service_name }} --type=LoadBalancer
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/expose_postgres_externally.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/get_superuser_secret.yml [hunk 23]
```diff
@@ -0,0 +1,7 @@
+---
+- name: Get the Cluster SuperUser Secret
+  shell: kubectl get secret -n {{ cnpg_namespace }} {{ cluster_superuser_name }} -oyaml -o=jsonpath={.data.password}|base64 -d
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/get_superuser_secret.yml:3]: `ansible.builtin.shell`

File: roles/manage_cnpg/tasks/get_superuser_secret.yml [hunk 24]
```diff
@@ -0,0 +1,7 @@
...
+  register: superusersecret
+
+- name: Display the Cluster SuperUser Secret
+  debug: var=superusersecret.stdout_lines
```
Comment (hannahms) [inline roles/manage_cnpg/tasks/get_superuser_secret.yml:7]: `ansible.builtin.debug`

**Review by hannahms — CHANGES_REQUESTED** [review]

File: roles/manage_cnpg/README.md (hunk 1 above)
Comment (dougortiz) [reply roles/manage_cnpg/README.md:94]: Correct, have updated.

File: manage_cnpg.yml (hunk 2 above)
Comment (dougortiz) [reply manage_cnpg.yml:98]: have removed unnecessary file

File: roles/manage_cnpg/files/cnp-sandbox.yml (hunk 3 above)
Comment (dougortiz) [reply roles/manage_cnpg/files/cnp-sandbox.yml:5]: this is correct

Comment (dougortiz) [reply roles/manage_cnpg/files/cnp-sandbox.yml:5]: updated

File: roles/manage_cnpg/tasks/maintenance_set.yml (hunk 5 above)
Comment (dougortiz) [reply roles/manage_cnpg/tasks/maintenance_set.yml:7]: updated

File: roles/manage_cnpg/tasks/expose_grafana_externally.yml (hunk 21 above)
Comment (dougortiz) [reply roles/manage_cnpg/tasks/expose_grafana_externally.yml:3]: updated

File: roles/manage_cnpg/tasks/drop_expose_postgres_externally.yml (hunk 20 above)
Comment (dougortiz) [reply roles/manage_cnpg/tasks/drop_expose_postgres_externally.yml:3]: updated

File: roles/manage_cnpg/tasks/expose_postgres_externally.yml (hunk 22 above)
Comment (dougortiz) [reply roles/manage_cnpg/tasks/expose_postgres_externally.yml:3]: updated

File: roles/manage_cnpg/tasks/get_superuser_secret.yml (hunk 24 above)
Comment (dougortiz) [reply roles/manage_cnpg/tasks/get_superuser_secret.yml:7]: updated

File: roles/manage_cnpg/files/cnp-sandbox.yml (hunk 3 above)
Comment (hannahms) [reply roles/manage_cnpg/files/cnp-sandbox.yml:5]: I don't see this change

**Pull request merged**